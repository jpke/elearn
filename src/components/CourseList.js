import React from 'react';
import {Link} from 'react-router';

const CourseList = (props) => {
  return (
    <div className="course-list">
      <h2>Select course</h2>
      <ul>
        {props.courses}
      </ul>
      {props.course &&
        <div>
          <p>Choose <Link to="/quiz" className="redirect">Quiz</Link> or <Link to="/lessons" className="redirect">Lessons</Link> to access course content</p>
          {props.course.admin &&
          <div>
            <p>Admin:</p>
            <Link to="/admin" className="redirect">
              <button className="manageUsers">Manage Enrolled Users</button>
            </Link>
          </div>
          }
        </div>
      }
      <button onClick={props.logOut}>Logout</button>
    </div>
  )
};

export default CourseList;
