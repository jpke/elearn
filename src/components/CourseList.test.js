/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import CourseList from './CourseList';
import CourseListCreator from '../utils/CourseListCreator'

let courses, passed, courseList;
beforeAll(() => {
  courses = [{
    name: "example course",
    id: "00000",
    enrollable: [{email: "new@email.com"}],
    quizzes: [
      {
        _id: "1111",
        title: "example Quiz"
      }
    ],
    admin: false
  }];

  passed = [{
    _id: "2222",
    of: "1110"
  }];

  courseList = CourseListCreator(courses, jest.fn(), passed, "someToken", "urlAddress");
})

it('renders correctly', () => {
  const tree = renderer.create(
    <CourseList
      courses={courseList}
      course={courses[0]}
      logOut={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
