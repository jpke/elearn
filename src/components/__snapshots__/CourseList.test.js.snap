exports[`test renders correctly 1`] = `
<div
  className="course-list">
  <h2>
    example course
  </h2>
  <ul>
    <li
      onClick={[Function]}>
      example course
    </li>
  </ul>
  <div>
    <p>
      Choose 
      <a
        className="redirect"
        onClick={[Function]}
        style={Object {}}>
        Quiz
      </a>
       or 
      <a
        className="redirect"
        onClick={[Function]}
        style={Object {}}>
        Lessons
      </a>
       to access course content
    </p>
  </div>
  <button
    onClick={[Function]}>
    Logout
  </button>
</div>
`;
